raiden/api/rest.py

```diff
@@ -1219,6 +1221,37 @@ def _set_channel_reveal_timeout(
         result = self.channel_schema.dump(updated_channel_state)
         return api_response(result=result)
 
+    def _set_fee_schedule(
+        self,
+        registry_address: TokenNetworkRegistryAddress,
+        channel_state: NettingChannelState,
+        fee_schedule: FeeScheduleState,
+    ):
+        log.debug("Set fee schedule")
+        if channel.get_status(channel_state) != ChannelState.STATE_OPENED:
+            return api_error(
+                errors="Can't update the fee schedule of a closed channel",
+                status_code=HTTPStatus.CONFLICT,
+            )
+        try:
+            self.raiden_api.set_fee_schedule(
+                registry_address=registry_address,
+                token_address=channel_state.token_address,
+                partner_address=channel_state.partner_state.address,
+                fee_schedule=fee_schedule,
+            )
+        except (NonexistingChannel, UnknownTokenAddress, InvalidBinaryAddress) as e:
+            return api_error(errors=str(e), status_code=HTTPStatus.BAD_REQUEST)
+        except InvalidFeeSchedule as e:
+            return api_error(errors=str(e), status_code=HTTPStatus.CONFLICT)
+
+        updated_channel_state = self.raiden_api.get_channel(
+            registry_address, channel_state.token_address, channel_state.partner_state.address
+        )
+
+        result = self.channel_schema.dump(updated_channel_state)
+        return api_response(result=result)
```

Comment:
You should not dump the schema manually. Return the channel state similar to other API handlers.

Reply:
I don't follow... This is following the exact same pattern as all of the private functions called inside `patch_channel`.